html/browsers/browsing-the-web/navigating-across-documents/replace-before-load/window-open-popup-during-load.html

```diff
@@ -0,0 +1,53 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/common/utils.js"></script>
+
+<body>
+<script>
+"use strict";
+promise_test(async t => {
+  // Generating a new one instead of hard-coding makes running manual tests a bit easier.
+  const windowName = token();
+
+  const code = `
+    window.onload = () => opener.navigateMe();
+  `;
+
+  const startURL = "resources/slow-code-injector.html?pipe=sub(none)&code=" + encodeURIComponent(code);
+  const absoluteStartURL = (new URL(startURL, location.href)).href;
+
+  const afterReplacementURL = "resources/message-opener.html";
+  const absoluteAfterReplacementURL = (new URL(afterReplacementURL, location.href)).href;
+
+  window.navigateMe = () => {
+    window.open(absoluteAfterReplacementURL, windowName);
+  };
+
+  const w = window.open(startURL, windowName);
+  t.add_cleanup(() => w.close());
+
+  // Wait to get past any initial about:blank
+  while (true) {
+    if (w.location.href === absoluteStartURL) {
+      break;
+    }
+    await new Promise(r => setTimeout(r, 0));
+  }
+
+  assert_equals(w.onloadFired, undefined, "onload must not yet have fired");
+  assert_equals(w.history.length, 1, "history.length for the opened window must start at 1");
+
+  await new Promise(r => { window.onmessage = r; });
+
+  assert_equals(w.history.length, 2, "history.length must increase");
+  assert_equals(w.location.href, absoluteAfterReplacementURL);
+
+  w.history.back();
+
+  t.step_timeout(() => {
+    asssert_equals(w.location.href, absoluteStartURL, "1 second after attempting to go back, it indeed went back");
+  }, 1000);
```

Comment:
Here and below: any way to shorten this? 1 second is a long time.

Reply:
Moved to waiting for a posted message